modules/claims_api/lib/bgs_service/manage_representative_service.rb

```diff
@@ -1,18 +1,53 @@
 # frozen_string_literal: true
 
-require_relative 'manage_representative_service/update_poa_request'
-
 module ClaimsApi
   class ManageRepresentativeService < ClaimsApi::LocalBGS
-    private
-
-    def make_request(**args)
-      super(
-        endpoint: 'VDC/ManageRepresentativeService',
-        namespaces: { 'data' => '/data' },
-        transform_response: false,
-        **args
-      )
+    def bean_name
+      'VDC/ManageRepresentativeService'
+    end
+
+    def read_poa_request(poa_codes: [])
+      # Workaround to allow multiple roots in the Nokogiri XML builder
+      # https://stackoverflow.com/a/4907450
+      doc = Nokogiri::XML::DocumentFragment.parse ''
+
+      Nokogiri::XML::Builder.with(doc) do |xml|
+        xml.send('data:POACodeList') do
+          poa_codes.each do |poa_code|
+            xml.POACode poa_code
+          end
+        end
+        xml.send('data:SecondaryStatusList') do
+          %w[New Pending Accepted Declined].each do |status|
+            xml.SecondaryStatus status
+          end
+        end
+      end
+
+      body = builder_to_xml(doc)
+
+      make_request(endpoint: bean_name, action: 'readPOARequest', body:, key: 'POARequestRespondReturnVO',
+                   namespaces: { 'data' => '/data' })
+    end
+
+    def update_poa_request(representative:, proc_id:)
+      first_name = representative.try(:first_name) || representative[:first_name]
+      last_name = representative.try(:last_name) || representative[:last_name]
+
+      builder = Nokogiri::XML::Builder.new do |xml|
+        xml.send('data:POARequestUpdate') do
+          xml.VSOUserFirstName first_name
+          xml.VSOUserLastName last_name
+          xml.dateRequestActioned Time.current.iso8601
+          xml.procId proc_id
+          xml.secondaryStatus 'obsolete'
+        end
+      end
+
+      body = builder_to_xml(builder)
+
+      make_request(endpoint: bean_name, action: 'updatePOARequest', body:, key: 'POARequestUpdate',
+                   namespaces: { 'data' => '/data' }, transform_response: false)
```

Comment:
Refactor of [`update_poa_request.rb`](https://github.com/department-of-veterans-affairs/vets-api/pull/18820/files#diff-44310c343f4239dce4fcc3ddb4d5ad03dc73fc2b135632103051fd6290b288cc).